articles/diagnose-diabetes-with-svm/index.md

```diff
@@ -0,0 +1,342 @@
+
+We get 0, which means this patient does not have diabetes. Let us view our test set.
+
+```python
+# Viewing Test Set
+X_test
+```
+
+![Test Set](/engineering-education/diagnose-diabetes-with-svm/test-set.jpg)
+
+Now let us try to diagnose the third patient in the test set (with id 113). Remember, the index of the third patient is two since we start counting from 0.
+
+```python
+# Checking the third patient in the test set with index 2
+X_test.iloc[2]
+```
+
+![Details of the Third Patient](/engineering-education/diagnose-diabetes-with-svm/third-patient-details.jpg)
+
+```python
+# Convert dataframe to a numpy array
+t_patient = np.array([ X_test.iloc[2]])
+
+# Predicting on third patient in Test Set
+t_patient = scaler.transform(t_patient)
+    
+print("Model's Prediction:", model.predict(t_patient))
+print("Actual Prediction:", y_test.iloc[2])
+```
+
+![Prediction of the Third Patient](/engineering-education/diagnose-diabetes-with-svm/model-prediction.jpg)
+
+We can see that our model prediction is 0, and the actual prediction is also 0. This means our model made the correct prediction for this patient. The third patient does not have diabetes.
+
+### Assess Model Performance
+Let us see the accuracy of the entire test set.
+
+```python
+# Accuracy on Testing Set
+X_test = scaler.transform(X_test)
+y_pred = model.predict(X_test)
+print("Accuracy Score:", accuracy_score(y_test, y_pred))
+```
+
+![Accuracy Score of the Test Set](/engineering-education/diagnose-diabetes-with-svm/accuracy-score.jpg)
+
+We normalize the test set before making predictions. We have an accuracy of 77.60, which is lower than what we had on the training set. This is because the test set contains data our model has not seen before.
+
+What if our model had predicted that no one had diabetes? What would be our accuracy in that case? Let us find out what the accuracy would be if our model had predicted 0 (no diabetes) for all the patients. To do this, we create an array of zeros with the same shape as our test set classes.
+
+```python
+# Comparison to All-Zero Prediction
+y_zero = np.zeros(y_test.shape)
+print(accuracy_score(y_test, y_zero))
+```
+
+![Accuracy Score of All-Zero Prediction](/engineering-education/diagnose-diabetes-with-svm/accuracy-of-all-zero-prediction.jpg)
+
+When we compare our test set with an all-zero array,  we get an accuracy score of 67.7%. Our model accuracy is 67.7%, even when it predicts that no one in the test set has diabetes. This means our dataset is unbalanced. There are more samples of the class without diabetes in our dataset. So accuracy score will not help us evaluate our model. We can measure the performance of our model by using Precision and Recall. 
+
+Precision tells us what fraction has diabetes from all the patients our model predicted to have diabetes. Recall gives us the fraction our model correctly detected as having diabetes out of all the diabetic patients.
```

Comment:
The definition you've included for Precision is great!
But please explain precision and recall in terms of True Positives and False Positives also. It would be a more systematic definition

Reply:
The formula of precision and recall and the definition of true positives, true negatives, false positives and false negatives has been added.